### DISCUSSION TITLE
RPi 4B vs 3B+

### DISCUSSION THREAD

**Discussion by Stimpson-J-Cat** [opencardev/crankshaft#478] [Q&A]
I had Crankshaft set up on a 3B+. I had a power supply set up on the accessory circuit in my truck that stayed powered for 7 min after shutdown. The Pi was connected to that. I had a relay set up that powered with ignition and set GPIO3 low (to ground from pin 9) and that would start the Pi, then when I shut the truck down, the relay would release, ground would drop off GPIO3 and the Pi would shut down, then 7 min later, the power supply would lose power and thus no power draw. That 3B+ died and I replaced it with a 4B and this set up no longer works. I even modified the wiring on my relay to match the Wiki and set GPIO3 low when ignition on and GPIO27 low when ignition off and all that happens is the Pi will boot and 10 seconds later, it shuts down (10 seconds is the delay I have set). It seems that any change in the state of GPIO3 makes the Pi boot; if I set it to low, it boots then shuts down in 10, and if I "unset" low (remove the ground from GPIO3) it boots again then shuts down in 10 seconds. I also had a similar relay set up to GPIO12 to dim the display at night when the lights were on and that no longer works on the Pi 4B either. Am I better off going back to a 3B+ and not using the 4B?

**Comment by matt2005** [comment]
Can you supply a copy of your wirring diagram for the circuit?

**Comment by Stimpson-J-Cat** [comment]
Sure. Drawing shows the wiring layout, but 85 and 86 are 12v from truck, 30 is connected to pin 9 on the Pi, 87 (NO) is connected to GPIO3 and 87a (NC) is connected to GPIO27.  The photo shows the actual connections.
![20210805_184232](https://user-images.githubusercontent.com/88469570/128437720-84d21a91-971b-47ef-b06a-2465ee49fd51.jpg)
![20210805_184223_HDR](https://user-images.githubusercontent.com/88469570/128437756-834f0896-442a-461d-ba3e-6760111beb4d.jpg)

**Reply by Stimpson-J-Cat** [reply]
I have tried 3 different versions of CS (8/19, 10/20, & 2/21) and all exhibit the same behavior. I will try the Pi3B+ next and see if it works there. I don't know what else to do outside of that, since this worked when I had it connected to a 3B+ and I have verified that my relays work. Anyone else have any suggestions?

**Reply by Stimpson-J-Cat** [reply]
In the crankshaft_env.sh file, it says that GPIO should stay low (GND) to keep the Pi running and go high (+5vdc) to initiate a shutdown after the IGNITION_DELAY. I can easily swap the pins around to do this. Will this solve my issue? Also, it says there that "Note: this gpio depends NOT on ENABLE_GPIO!!!" Does this mean I should have "ENABLE_GPIO=0" or set to "1" to enable it or does it not matter the state of that flag at all? It's a little confusing.

**Reply by matt2005** [reply]
I've not got access to my car with pi for a week so I cant check till I get it back

**Reply by Stimpson-J-Cat** [reply]
I can 100% confirm that everything works as it should on the 3B but not the 4B. With the latest version of CS and wired the way my diagrams depict, the system boots and shuts down properly when when powering the ignition relay on GPIO3 and the dimmer works when powering the relay on GPIO12.  None of this works properly on the 4B. Also found that if I didn't have the dimmer set up on the CS ENV page, the day/night button on screen would cause the screen to go completely black but was still responsive - if I hit the screen where the button would be, it would come back on.

SO - I am sticking with the 3B for now.

**Reply by matt2005** [reply]
I must admit that i use a 3B, i've only recently got a 4 8GB but thats currently used for other purposes.
Can you raise an issue for this?

**Reply by Stimpson-J-Cat** [reply]
Done!

**Reply by matt2005** [reply]
This should be fixed in alpha 7.4, the issue was the gpio library. I've updated it and tested on my pi4

**Comment by matt2005** [comment]
Resolved in Alpha 7.4
